tests/unit/deploy/test_deployer.py

```diff
@@ -854,6 +855,82 @@ def test_can_update_auth_handlers(self, sample_app_with_auth):
             zip_contents=b'package contents',
         )
 
+    def test_can_create_auth_with_different_config(self, sample_app_with_auth):
+        # We're notusing create_config_obj because we want to approximate
+        # loading config from disk which contains per-lambda configuration.
+        disk_config = {
+            'app_name': 'myapp',
+            'iam_role_arn': 'role-arn',
+            'manage_iam_role': False,
+            'stages': {
+                'dev': {
+                    'lamba_timeout': 10,
+                    'lambda_functions': {
+                        'myauth': {
+                            'lambda_timeout': 20,
+                            'lambda_memory_size': 512,
+                        }
+                    }
+                }
+            }
+        }
+        config = Config(
+            'dev',
+            config_from_disk=disk_config,
+            user_provided_params={'chalice_app': sample_app_with_auth,
+                                  'project_dir': '.'}
+        )
+        deployer = LambdaDeployer(
+            self.aws_client, self.packager, None, self.osutils,
+            self.app_policy)
+        self.aws_client.lambda_function_exists.return_value = False
+        self.aws_client.create_function.side_effect = [
+            self.lambda_arn, 'arn:auth-function']
+        deployer.deploy(config, None, stage_name='dev')
+        self.aws_client.create_function.assert_called_with(
```

Comment:
So right now we are only checking the last client `create_function` call which was for the auth handler. The first `create_function` call is for the api handler which is not checked, but has different parameters such as its lambda timeout will be 10 and its lambda memory size will be the default. The assertion would be on what was called in the first `create_function` call and it would be just a sanity check to make sure the correct scope was used in creating the api handler. If there is already a check for that though, I don't think we need to add an assertion. I may have missed it if there is one as I am only looking at the diffs as you add them, not the entire PR diff each time.